analyzers/rolaguard_base_analyzer/ResourceMeter.py

```diff
@@ -30,24 +30,26 @@ def __call__(self, asset, packet):
 
         if type(asset) == Device:
             if asset.id in self.device_stats: # entry for this device already created in device stats dict, update values
-                if self.device_resource_usage(asset, packet):
+                if self.device_resource_usage(asset, packet, policy_manager):
                     if packet.uplink:
                         self.device_stats[asset.id]["last_fcount"] = packet.f_count
                     else:
                         self.device_stats[asset.id]["last_fcount_down"] = packet.f_count
-                    self.device_stats[asset.id]["last_date"] = packet.date
+                    self.device_stats[asset.id]["last_date"][packet.gateway] = packet.date
             else: # initialize device_stats dict entry with default values
                 self.device_stats[asset.id] = {
                     "last_fcount" : packet.f_count if packet.uplink else None,
                     "last_fcount_down" : packet.f_count if not packet.uplink else None,
-                    "last_date" : packet.date,
-                    "rssi" : {},    # dict containing the gateway hex ids in which the device is connected as key, and rssi numbers as values 
-                    "lsnr" : {}     # dict containing the gateway hex ids in which the device is connected as key, and lsnr numbers as values 
+                    "last_date" : {},   # dict containing the gateway hex ids in which the device is connected as key, and most recent date of packet as values 
+                    "rssi" : {},        # dict containing the gateway hex ids in which the device is connected as key, and rssi numbers as values 
+                    "lsnr" : {}         # dict containing the gateway hex ids in which the device is connected as key, and lsnr numbers as values 
                 }
-                if packet.rssi is not None:
+                if packet.rssi:
                     self.device_stats[asset.id]["rssi"][packet.gateway] = packet.rssi
-                if packet.lsnr is not None:
+                if packet.lsnr:
```

Comment:
I think RSSI it is not a problem, but LSNR could be 0. It is safer if it is ask "is not None"

Reply:
You're right, let me change it back